.circleci/config.yml

```diff
@@ -310,14 +310,9 @@ defaults:
         CXX: clang++
         MAKEFLAGS: -j 3
 
-  - base_ubuntu2004_clang_small: &base_ubuntu2004_clang_small
+  - base_ubuntu2004_clang_medium: &base_ubuntu2004_clang_medium
       <<: *base_ubuntu2004_clang
-      resource_class: small
-      environment:
```

Comment:
I am conflicted. This is exactly what you should do (deleting redundant/inherited fields) even though it impacts readability, but we seem to be using the redundancy approach all over this file when defining containers. So if readability is of paramount importance, then we might as well not use a base definition of the container/step/whatever if we're gonna overwrite all of the fields with the same values again.

Reply:
@nishant-sachdeva Please restore the environment section. Providing values for these variables is part of the point of having these base definitions here.